chringo/lib/src/pages/email_login_page.dart

```diff
@@ -0,0 +1,772 @@
+            },
+          ),
+          Container(
+            padding: const EdgeInsets.symmetric(vertical: 16.0),
+            alignment: Alignment.center,
+            child: RaisedButton(
+              onPressed: () async {
+                if (_formKey.currentState.validate()) {
+                  _signInWithEmailAndPassword();
+                }
+              },
+              child: const Text('Submit'),
+            ),
+          ),
+          Container(
+            alignment: Alignment.center,
+            padding: const EdgeInsets.symmetric(horizontal: 16),
+            child: Text(
+              _success == null
+                  ? ''
+                  : (_success
+                      ? 'Successfully signed in ' + _userEmail
+                      : 'Sign in failed'),
+              style: TextStyle(color: Colors.red),
+            ),
+          )
+        ],
+      ),
+    );
+  }
+
+  @override
+  void dispose() {
+    _emailController.dispose();
+    _passwordController.dispose();
+    super.dispose();
+  }
+
+  // Example code of how to sign in with email and password.
+  void _signInWithEmailAndPassword() async {
+    final FirebaseUser user = (await _auth.signInWithEmailAndPassword(
+      email: _emailController.text,
+      password: _passwordController.text,
+    ))
+        .user;
+    if (user != null) {
+      setState(() {
+        _success = true;
+        _userEmail = user.email;
+      });
+    } else {
+      _success = false;
+    }
+  }
+}
+
+class _EmailLinkSignInSection extends StatefulWidget {
+  @override
+  State<StatefulWidget> createState() => _EmailLinkSignInSectionState();
+}
+
+class _EmailLinkSignInSectionState extends State<_EmailLinkSignInSection>
+    with WidgetsBindingObserver {
+  final GlobalKey<FormState> _formKey = GlobalKey<FormState>();
+  final TextEditingController _emailController = TextEditingController();
+
+  bool _success;
+  String _userEmail;
+  String _userID;
+
+  @override
+  void initState() {
+    super.initState();
+    WidgetsBinding.instance.addObserver(this);
+  }
+
+  @override
+  void dispose() {
+    _emailController.dispose();
+    WidgetsBinding.instance.removeObserver(this);
+    super.dispose();
+  }
+
+  @override
+  void didChangeAppLifecycleState(AppLifecycleState state) async {
+    if (state == AppLifecycleState.resumed) {
+      final Uri link = await _retrieveDynamicLink();
+
+      if (link != null) {
+        final FirebaseUser user = (await _auth.signInWithEmailAndLink(
+          email: _userEmail,
+          link: link.toString(),
+        ))
+            .user;
+
+        if (user != null) {
+          _userID = user.uid;
+          _success = true;
+        } else {
+          _success = false;
+        }
+      } else {
+        _success = false;
+      }
+
+      setState(() {});
+    }
+  }
+
+  Future<Uri> _retrieveDynamicLink() async {
+    final PendingDynamicLinkData data =
+        await FirebaseDynamicLinks.instance.retrieveDynamicLink();
```

Comment:
retrieveDynamicLink is not defined as a method for this class

Reply:
https://firebase.google.com/docs/dynamic-links/android/receive#set-up-firebase-and-the-dynamic-links-sdk